stravalib/model.py

```diff
@@ -432,379 +553,165 @@ class ActivityKudos(Athlete):
-    elevation_difference = Attribute(
-        float, units=uh.meters
-    )  #: Elevation difference for split
-    moving_time = (
-        TimeIntervalAttribute()
-    )  #: :class:`datetime.timedelta` of moving time for split
-    average_heartrate = Attribute(float)  #: Average HR for split
-    split = Attribute(int)  #: Which split number
-    pace_zone = Attribute(int)  #: (undocumented)
-    average_speed = Attribute(float, units=uh.meters_per_second)
-    average_grade_adjusted_speed = Attribute(float, units=uh.meters_per_second)
+    _unit_registry = {
+        'distance': 'meters',
+        'elevation_difference': 'meters',
+        'average_speed': 'm/s'
+    }
 
-    def __repr__(self):
-        return "<Split split={} distance={} elapsed_time={}>".format(
-            self.split, self.distance, self.elapsed_time
-        )
+    _field_conversions = {'elapsed_time': time_interval, 'moving_time': time_interval}
 
 
-class SegmentExplorerResult(LoadableEntity):
+class SegmentExplorerResult(
+    ExplorerSegment, BackwardCompatibilityMixin, DeprecatedSerializableMixin, BoundClientEntity
+):
     """
     Represents a segment result from the segment explorer feature.
 
     (These are not full segment objects, but the segment object can be fetched
     via the 'segment' property of this object.)
     """
 
-    _segment = None
-    id = Attribute(int)  #: ID of the segment.
-    name = Attribute(str)  #: Name of the segment
-    climb_category = Attribute(
-        int
-    )  #: Climb category for the segment (0 is higher)
-    climb_category_desc = Attribute(str)  #: Climb category text
-    avg_grade = Attribute(float)  #: Average grade for segment.
-    start_latlng = LocationAttribute()  #: Start lat/lon for segment
-    end_latlng = LocationAttribute()  #: End lat/lon for segment
-    elev_difference = Attribute(
-        float, units=uh.meters
-    )  #: Total elevation difference over segment.
-    distance = Attribute(float, units=uh.meters)  #: Distance of segment.
-    points = Attribute(str)  #: Encoded Google polyline of points in segment
-    starred = Attribute(
-        bool
-    )  #: Whether this segment is starred by authenticated athlete
+    _unit_registry = {
+        'elev_difference', 'meters',
+        'distance', 'meters'
+    }
 
-    @property
+    _latlng_extensions = extend_types('start_latlng', 'end_latlng', model_class='LatLon')
+
+    @lazy_property
     def segment(self):
         """Associated (full) :class:`stravalib.model.Segment` object."""
-        if self._segment is None:
-            self.assert_bind_client()
-            if self.id is not None:
-                self._segment = self.bind_client.get_segment(self.id)
-        return self._segment
+        return self.bound_client.get_segment(self.id)
 
 
-class AthleteSegmentStats(BaseEntity):
+class AthleteSegmentStats(SummarySegmentEffort, BackwardCompatibilityMixin, DeprecatedSerializableMixin):
```

Comment:
it looks like segments was broken into stats and streams?
so count is no longer a part of the stats? 

https://developers.strava.com/docs/reference/#api-Streams-getSegmentEffortStreams

Reply:
I haven't touched streams yet, so I don't know.

Reply:
ok no problem. !